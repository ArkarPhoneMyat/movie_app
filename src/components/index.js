import CustomFlatlist from "./custom_flatlist";

export {CustomFlatlist}