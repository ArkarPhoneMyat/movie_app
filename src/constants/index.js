import {myColor} from './my_color';
import {icons} from './icons';
import { sizes } from './sizes';

export {myColor, icons, sizes};
