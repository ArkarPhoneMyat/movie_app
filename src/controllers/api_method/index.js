import { getApi } from "./get_api";

export {getApi}