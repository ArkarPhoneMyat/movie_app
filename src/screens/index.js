import HomeScreen from './home/HomeScreen';
import DetailScreen from './detail/DetailScreen';
import SearchScreen from './search/SearchScreen';

export {HomeScreen, DetailScreen, SearchScreen};
