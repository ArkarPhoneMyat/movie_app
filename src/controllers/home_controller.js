import {getApi} from './api_method'

const getMovies = (setData) => {
    getApi('movies', data => {
        setData(data);
    })
};

export const homeController = {getMovies}