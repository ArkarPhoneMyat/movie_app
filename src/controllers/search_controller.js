import { getApi } from "./api_method";

const getMoviesSearch = (text, setData) => {
    getApi(`movies?q=${text}`, data => {
        setData(data);
    })
};

export const searchController = {getMoviesSearch}