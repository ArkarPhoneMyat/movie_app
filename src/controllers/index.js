import { homeController } from "./home_controller";
import { searchController } from "./search_controller";

export {homeController, searchController}