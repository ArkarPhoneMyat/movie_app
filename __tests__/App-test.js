import React from 'react';
import {describe, expect, test} from '@jest/globals';
import { homeController, searchController } from '../src/controllers';
import renderer from 'react-test-renderer'
import {HomeScreen} from '../src/screens';

describe('App testing', () => {
    test('renderer', () => {
        const screen = renderer.create(<HomeScreen/>).toJSON();
        expect(screen).toMatchSnapshot();
    })
    test('get Movies Api', () => {
        homeController.getMovies(data => {
            expect(data.length).toHaveLength();
        })
    })
    test('get Movies Search Api', () => {
        searchController.getMoviesSearch('Man', data => {
            expect(data.length).toHaveLength();
        })
    })
});