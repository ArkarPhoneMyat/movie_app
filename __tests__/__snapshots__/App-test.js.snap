// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App testing renderer 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 30,
              "fontWeight": "bold",
            }
          }
        >
          WOOKIE
        </Text>
        <Text
          style={
            Object {
              "fontSize": 30,
              "fontWeight": "bold",
            }
          }
        >
          MOVIES
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginBottom": 10,
            "marginTop": 20,
          }
        }
      >
        <View
          style={
            Object {
              "height": 250,
              "marginVertical": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "marginLeft": 10,
              }
            }
          >
            Action
          </Text>
          <RCTScrollView
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "height": 250,
              "marginVertical": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "marginLeft": 10,
              }
            }
          >
            Adventure
          </Text>
          <RCTScrollView
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "height": 250,
              "marginVertical": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "marginLeft": 10,
              }
            }
          >
            Animation
          </Text>
          <RCTScrollView
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "height": 250,
              "marginVertical": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "marginLeft": 10,
              }
            }
          >
            Biography
          </Text>
          <RCTScrollView
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "height": 250,
              "marginVertical": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "marginLeft": 10,
              }
            }
          >
            Crime
          </Text>
          <RCTScrollView
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "height": 250,
              "marginVertical": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "marginLeft": 10,
              }
            }
          >
            Drama
          </Text>
          <RCTScrollView
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "height": 250,
              "marginVertical": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "marginLeft": 10,
              }
            }
          >
            Family
          </Text>
          <RCTScrollView
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "height": 250,
              "marginVertical": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "marginLeft": 10,
              }
            }
          >
            History
          </Text>
          <RCTScrollView
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "height": 250,
              "marginVertical": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "marginLeft": 10,
              }
            }
          >
            Mystery
          </Text>
          <RCTScrollView
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "height": 250,
              "marginVertical": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "marginLeft": 10,
              }
            }
          >
            Romance
          </Text>
          <RCTScrollView
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "height": 250,
              "marginVertical": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "marginLeft": 10,
              }
            }
          >
            Sci-Fi
          </Text>
          <RCTScrollView
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "height": 250,
              "marginVertical": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "marginLeft": 10,
              }
            }
          >
            Thriller
          </Text>
          <RCTScrollView
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "height": 250,
              "marginVertical": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "marginLeft": 10,
              }
            }
          >
            War
          </Text>
          <RCTScrollView
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
